packages/devtools_app/lib/src/utils.dart

```diff
@@ -1305,3 +1281,18 @@ extension BoolExtension on bool {
     return -1;
   }
 }
+
+Future<T> whenValueNonNull<T>(ValueListenable<T> listenable) {
+  if (listenable.value != null) return Future.value(listenable.value);
+  final completer = Completer<T>();
+  void listener() {
+    final value = listenable.value;
+    if (value != null) {
+      completer.complete(value);
+    }
+    listenable.removeListener(listener);
```

Comment:
what if `value` is null? Then we would remove the listener and the completer would never complete. If `value` is null, should we keep listening?

Reply:
agreed. Fixed the logic.